src/contexts/GlobalContexts.tsx

```diff
@@ -15,41 +15,52 @@ import { UserDataContextProvider } from './UserDataContext';
 import { ExploreContextProvider } from './ExploreContext';
 import { ENSAddressContextProvider } from './ENSAddressContext';
+import { GraphDataContextProvider } from './GraphDataContext';
+import { DataLoadingContextProvider } from './DataLoadingContext';
+import { TokenBalanceContextProvider } from './TokenBalanceContext';
 
 export const GlobalContexts = (props: { children: React.ReactNode }) => {
     return (
         <AppStateContextProvider>
             <ENSAddressContextProvider>
                 <CachedDataContextProvider>
-                    <CrocEnvContextProvider>
-                        <TokenContextProvider>
-                            <ChainDataContextProvider>
-                                <UserDataContextProvider>
-                                    <ChartContextProvider>
-                                        <RangeContextProvider>
-                                            <TradeTokenContextProvider>
-                                                <PoolContextProvider>
-                                                    <CandleContextProvider>
-                                                        <TradeTableContextProvider>
-                                                            <UserPreferenceContextProvider>
-                                                                <SidebarContextProvider>
-                                                                    <ExploreContextProvider>
-                                                                        {
-                                                                            props.children
-                                                                        }
-                                                                    </ExploreContextProvider>
-                                                                </SidebarContextProvider>
-                                                            </UserPreferenceContextProvider>
-                                                        </TradeTableContextProvider>
-                                                    </CandleContextProvider>
-                                                </PoolContextProvider>
-                                            </TradeTokenContextProvider>
-                                        </RangeContextProvider>
-                                    </ChartContextProvider>
-                                </UserDataContextProvider>
-                            </ChainDataContextProvider>
-                        </TokenContextProvider>
-                    </CrocEnvContextProvider>
+                    <DataLoadingContextProvider>
+                        <UserDataContextProvider>
+                            <ChartContextProvider>
+                                <RangeContextProvider>
+                                    {/* Everything above here has no context dependencies */}
```

Comment:
Here is where I looked through contexts and placed those with no dependencies on other contexts above this line.